src/CTA.WebForms2Blazor/FileConverters/ViewFileConverter.cs

```diff
@@ -1,26 +1,93 @@
+        private List<(HtmlNode Node, HtmlNode Parent, ControlConverter Rules)> NodesToBeConverted;
+        private string _relativeDirectory;
+
+        public ViewFileConverter(string sourceProjectPath, string fullPath, Dictionary<String, ControlConverter> dict) 
+            : base(sourceProjectPath, fullPath)
         {
             // TODO: Register file with necessary services
+            _relativeDirectory = Path.GetDirectoryName(RelativePath);
+            NodesToBeConverted = new List<(HtmlNode Node, HtmlNode Parent, ControlConverter Rules)>();
+            ControlRulesMap = dict;
+        }
+
+        private HtmlDocument GetRazorContents()
+        {
+            var htmlDoc = new HtmlDocument();
+            htmlDoc.Load(FullPath);
+            DfsPostorderWalker(htmlDoc.DocumentNode, null);
+            
+            //This will either return string contents or modified HtmlDocument
+            //that will then be changed to a file information object
+            
+            ConvertNodes();
+            return htmlDoc;
+        }
+
+        private void DfsPostorderWalker(HtmlNode node, HtmlNode parent)
+        {
+            if (node == null)
+            {
+                return;
+            }
+
+            foreach (HtmlNode child in node.ChildNodes)
+            {
+                DfsPostorderWalker(child, node);
+            }
+
+            GetRules(node, parent);
+        }
+
+        private void GetRules(HtmlNode node, HtmlNode parent)
+        {
+            if (ControlRulesMap.ContainsKey(node.Name))
+            {
+                var conversionTuple = (node, parent, ControlRulesMap[node.Name]);
+                NodesToBeConverted.Add(conversionTuple);
+            }
+        }
+
+        private void ConvertNodes()
+        {
+            foreach (var package in NodesToBeConverted)
+            {
+                HtmlNode convertedNode = package.Rules.Convert2Blazor(package.Node);
+                package.Parent.ReplaceChild(convertedNode, package.Node);
+            }
         }
 
         public override async Task<IEnumerable<FileInformation>> MigrateFileAsync()
         {
             // TODO: Store UI information in necessary services
 
-            // View file converters don't need to return any file
-            // information objects, this is to be handled by the
-            // code-behinds for each
-            return Enumerable.Empty<FileInformation>();
+            // View file converters will return razor file contents with
+            // only view layer, code behind will be created in another file
+            HtmlDocument migratedDocument = GetRazorContents();
+            string contents = migratedDocument.DocumentNode.WriteTo();
+
+            string newFileName = "testing.razor"; //new file name depends, add later
+            string newPath = Path.Combine(_relativeDirectory, newFileName);
```

Comment:
These look like artifacts from testing. Can we remove these and update this method to be closer to a production-ready form?